File: python/invoke_model.py
import boto3
import json

MODEL_ID = "anthropic.claude-3-haiku-20240307-v1:0"

client = boto3.client("bedrock-runtime", region_name="us-east-1")

request_body = {
    "anthropic_version": "bedrock-2023-05-31",
    "messages": [
        {
            "role": "user",
            "content": [
                {
                    "type": "text",
                    "text": "What is 'rubber duck debugging'?"
                }
            ]
        }
    ],
    "max_tokens": 500,
}

response = client.invoke_model(
    modelId=MODEL_ID,
    body=json.dumps(request_body)
)

response_body = json.loads(response["body"].read())

response_text = response_body["content"][0]["text"]

print(response_text)

File: javascript/invokeModel/invokeModel.js
import { BedrockRuntimeClient, InvokeModelCommand } from "@aws-sdk/client-bedrock-runtime";

const region = "us-east-1";

const modelId = "meta.llama3-8b-instruct-v1:0";

const prompt = "Explain 'rubber duck debugging'";

const client = new BedrockRuntimeClient({ region });

const formattedPrompt = `
<|begin_of_text|><|start_header_id|>user<|end_header_id|>
${prompt}
<|eot_id|>
<|start_header_id|>assistant<|end_header_id|>
`;

const nativeRequest = {
    prompt: formattedPrompt,
    temperature: 0.5,
    max_gen_len: 512
};

const request = {
    modelId,
    body: JSON.stringify(nativeRequest),
    contentType: "application/json"
};

try {
    const command = new InvokeModelCommand(request);
    const response = await client.send(command);

    const nativeResponse = JSON.parse(new TextDecoder().decode(response.body));

    const text = nativeResponse.generation;
    console.log(text);
} catch (error) {
    console.error("\nError:", error);
}